permission/core/cache.go

```diff
@@ -1,22 +1,48 @@
-package types
+package core
```

Comment:
I would change the package name to `permcore` to disambiguate the codebase to not confuse between `core` go-ethereum package and `permission/core`

Reply:
Since this package has the core logic for permissions, feel `permission/core` is fine. We have similar examples e.g. `signer/core` and `consensus/istanbul/core`.